### DISCUSSION TITLE
Dietpi backup with docker containers running

### DISCUSSION THREAD

**Discussion by berndh1979** [MichaIng/DietPi#6814] [Software Request]
I run the dietpi on a PC Native SBC. Really enjoy this piece of software with all the features included

One request though: If I use the dietpi-backup tooling to backup my dietpi to a nfs share, it stops the docker containers. I have a important mail server runnig in docker container, that I don't want to stop/start while doing a backup. 

Would it be possible to backup the dietpi software to a nfs without stopping docker, hence the mailserver container keeps running?

Thanks

**Comment by Joulinar** [comment]
Yes, simply go to `dietpi-services` and exclude Docker service from being DietPi controlled.

**Reply by berndh1979** [reply]
Indeed, this works indeed. However, does this still start the excluded services after reboot? Sorry I'm not familiar yet with the dietpi-services tool.

**Reply by MichaIng** [reply]
Yes, "Mode" enabled/disabled/mask controls whether it is started at boot or able to start at all. "Include/Exclude" controls whether services are stopped/restarted before/after maintenance tasks, updates, backups and software installations. Let me know whether the texts in the dialogues could be clearer.

Keep in mind that, if `/mnt/dietpi_userdata` is left on the rootfs, all userdata, including Docker images and volumes (at their default dir at least), are backed up as well, which might cause problems/inconsistencies when containers are currently running and generating data. So in this case, better exclude the directory in this case (remove the `+ /mnt/dietpi_usedata` line from filter).

**Reply by berndh1979** [reply]
Great work. it all works nicely. However, I don't want to exclude the /mnt/dietpi_userdata actually as this will hold important files to backup, like mysql files and other important software settings/data that are installed. 

Having the dockers run in parallel with the backup and docker updates the files is OK for me though. I did not see an issue with that (yet). 

I see the speed of the backup is somewhat low (35-40 MB/sec) on a 2.5Gbit link to my server. If I put my dietpi device to Performance governor, it still keeps this speed. I also do not see one core to max out (100% utilization). Any suggestions to speed this up?

**Reply by MichaIng** [reply]
You could also exclude the `docker-data` directory only, and instead create a dedicated job for it. Many containers/applications also have their internal backup mechanisms, which are more reliable to use during operation compared to copying raw data.

NFS is known to be somewhat slow, especially when it is about a large number of smaller files. You could benchmark the NFS mount, like
```sh
FP_BENCHFILE=/mnt/nfs_client BENCH_FILESIZE=100 /boot/dietpi/func/dietpi-benchmark 1
```
